### PR TITLE
Support config options via command line

### PR DESCRIPTION
**Pull request by aomader**
This is going to be a fix for #49.

Though, it's still not clear to me what is meant by "disabling permissions". Could you elaborate what to do exactly?

### CODE REVIEW

**Comment by Razican** [comment]
Permissions would be to disable permission checks, one by one, but we are still not sure if we would like to include labels or some sort of identifiers. So, for now, no permision check modifications through CLI :)

File: src/main.rs [hunk 1]
```diff
@@ -61,23 +61,55 @@ fn main() {
...
 
+    config.set_app_package(app_package);
+    config.set_verbose(verbose);
+    config.set_quiet(quiet);
+    config.set_force(force);
+    config.set_bench(bench);
+    config.set_open(open);
+
+    if let Some(threads) = matches.value_of("threads") {
+        config.set_threads(threads.parse().expect("An integer"));
```
Comment (Razican) [inline src/main.rs]: This actually should be an integer smaller than `u8::MAX`.

File: src/main.rs [hunk 2]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Path to the jd-cmd file.")
+            .takes_value(true))
+        .arg(Arg::with_name("results-template")
+            .short("T")
+            .long("results-template")
+            .help("Path to a results template file.")
+            .takes_value(true))
+        .arg(Arg::with_name("rules-json")
+            .short("R")
+            .long("rules-json")
```
Comment (Razican) [inline src/main.rs]: As said above, no short option, and a `--rules` option would be enough.

File: src/main.rs [hunk 3]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Path to a JSON rules file.")
+            .takes_value(true))
+        .arg(Arg::with_name("disable-permissions")
```
Comment (Razican) [inline src/main.rs]: As I said in the comment, I think is better to remove permissions for now.

File: src/main.rs [hunk 4]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
         .arg(Arg::with_name("open")
             .long("open")
             .help("Open the report in a browser once it is complete."))
+        .arg(Arg::with_name("threads")
+            .short("t")
+            .long("threads")
+            .help("Number of threads to use.")
+            .takes_value(true))
+        .arg(Arg::with_name("downloads-folder")
```
Comment (Razican) [inline src/main.rs]: Here `--downloads` should be enough., and I would not add the short option.

File: src/main.rs [hunk 5]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .short("d")
+            .long("downloads-folder")
+            .help("Folder where the downloads are stored.")
+            .takes_value(true))
+        .arg(Arg::with_name("dist-folder")
+            .short("D")
```
Comment (Razican) [inline src/main.rs]: Same as before `--dist` should be enough, and without the short option.

File: src/main.rs [hunk 6]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .long("dist-folder")
+            .help("Distribution folder.")
```
Comment (Razican) [inline src/main.rs]: Maybe a description such as _"Folder where distribution files will be extracted"_ would be better.

File: src/main.rs [hunk 7]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .takes_value(true))
+        .arg(Arg::with_name("results-folder")
+            .short("r")
+            .long("results-folder")
```
Comment (Razican) [inline src/main.rs]: Same as above, no short option and only `--results` instead of `--results-folder`.

File: src/main.rs [hunk 8]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Folder where to store the results.")
+            .takes_value(true))
+        .arg(Arg::with_name("apktool-file")
+            .short("a")
+            .long("apktool-file")
```
Comment (Razican) [inline src/main.rs]: Same as above, without the short option and only `--apktool`.

File: src/main.rs [hunk 9]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Path to the apktool file.")
+            .takes_value(true))
+        .arg(Arg::with_name("dex2jar-folder")
+            .short("j")
+            .long("dex2jar-folder")
```
Comment (Razican) [inline src/main.rs]: Same as above, no short option and `--dex2jar` only.

File: src/main.rs [hunk 10]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Where to store the jar files.")
+            .takes_value(true))
+        .arg(Arg::with_name("jd-cmd-file")
+            .short("J")
+            .long("jd-cmd-file")
```
Comment (Razican) [inline src/main.rs]: As before, no short option and `--jd-cmd` only.

File: src/main.rs [hunk 11]
```diff
@@ -365,6 +397,57 @@ fn get_help_menu() -> ArgMatches<'static> {
...
+            .help("Path to the jd-cmd file.")
+            .takes_value(true))
+        .arg(Arg::with_name("results-template")
+            .short("T")
```
Comment (Razican) [inline src/main.rs]: This will be changed with the new templating system, but as before, no short option and `--template` as the option.

**Review by Razican — CHANGES_REQUESTED** [review]
Check the changes I proposed. Could you also add some documentation to setters?

Thanks for the pull request, in any case, looks great!

File: src/main.rs (hunk 1 above)
Comment (aomader) [reply src/main.rs]: `parse()` ensures this, however in a not very userfriendly way. How should I instead do this?

**Comment by aomader** [comment]
@Razican Done.

**Review by Razican — APPROVED** [review]

**Comment by Razican** [comment]
Looks great! I'll merge as soon as tests pass, thanks!

**Pull request merged**